src/improc.c

```diff
@@ -2512,3 +2512,175 @@ DoubleImage ifft2DDouble(ComplexImage image) {
 
   return im;
 }
+
+typedef struct quack {
+  int *buffer;
+  int start;
+  int end;
+  int size;
+  int capacity;
+} Quack;
+
+inline int quackIsEmpty(Quack *quack) {
+  return quack->size == 0;
+}
+
+inline int quackPeekBack(Quack *quack) {
+  return quack->buffer[(quack->end + quack->capacity - 1) % quack->capacity];
+}
+
+static int quackPopBack(Quack *quack) {
+  int value = quackPeekBack(quack);
+  quack->end = (quack->end + quack->capacity - 1) % quack->capacity;
+  quack->size--;
+  return value;
+}
+
+static void quackPushBack(Quack *quack, int value) {
+  if (quack->size >= quack->capacity) {
+    fatalError("attempted to insert into full quack. capactiy: %d", quack->capacity);
+  }
+  quack->buffer[quack->end] = value;
+  quack->end = (quack->end + 1) % quack->capacity;
+  quack->size++;
+}
+
+inline int quackPeekFront(Quack *quack) {
+  return quack->buffer[quack->start];
+}
+
+static int quackPopFront(Quack *quack) {
+  int value = quackPeekFront(quack);
+  quack->start = (quack->start + 1) % quack->capacity;
+  quack->size--;
+  return value;
+}
+
+#if 0
+// this function is unused, but i don't know if we want to keep it for completion sake
+static void quackPushFront(Quack *quack, int value) {
+  if (quack->size >= quack->capacity) {
+    fatalError("attempted to insert into full quack. capactiy: %d", quack->capacity);
+  }
+  quack->start = (quack->start + quack->capacity - 1) % quack->capacity;
+  quack->buffer[quack->start] = value;
+  quack->size++;
+}
+#endif
```

Comment:
I commented this out with #if 0 because we don't use this function.. However, it makes the quack implementation more complete if we have this. What do you think, keep this?

Reply:
Might indeed be nice to keep it for completeness sake

Reply:
The problem is we get compilation warnings